File: 1301-1400/1313-decompress-rle-list.py
class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        i = 0
        result = []
        while i < len(nums):
            for j in range(nums[i]):
                result.append(nums[i+1])
            i += 2
        return result

File: 1301-1400/1313-decompress-rle-list.java
class Solution {
  public int[] decompressRLElist(int[] nums) {
      ArrayList<Integer> result = new ArrayList<Integer>();
      for(int i=0;i<nums.length;i+=2) {
          int freq = nums[i];
          int val = nums[i+1];
          for(int j=0;j<freq;j++) {
              result.add(val);
          }
      }
      
      return result.stream().mapToInt(i -> i).toArray();
  }
}